day3/m_partitioner.F90

```diff
@@ -5,13 +5,17 @@
 !> @author Michal Sudwoj
 !> @date 2020-06-15
 module m_partitioner
-  use, intrinsic :: iso_fortran_env, only: REAL32, REAL64
-  use mpi!, only: MPI_Comm_Rank, MPI_Comm_Size, MPI_Scatter, MPI_Gather, MPI_Allgather, MPI_FLOAT
+  use, intrinsic :: iso_fortran_env, only: REAL32, REAL64, error_unit
+  use mpi, only: &
```

Comment:
`use only` avoids namespace conflicts

Reply:
Sadly, MPICH is non-conforming here, and does not expose `MPI_Scatter`, `MPI_Gather`, `MPI_Allgather` or any other type-generic MPI function. Added comment to clarify why those are commented out.